custom_components/smartthinq_sensors/binary_sensor.py

```diff
@@ -343,18 +342,12 @@ def assumed_state(self) -> bool:
         """Return True if unable to access real state of the entity."""
         return self._api.assumed_state
 
-    def _get_on_state(self):
+    def _get_on_state(self) -> bool:
         """Return true if the binary sensor is on."""
-        ret_val = self._get_sensor_state()
-        if ret_val is None:
-            return False
-        if isinstance(ret_val, bool):
-            return ret_val
-        ret_val = ret_val.lower()
-        if ret_val == STATE_ON:
-            return True
-        state = STATE_LOOKUP.get(ret_val, STATE_OFF)
-        return state == STATE_ON
+        sensor_state = self._get_sensor_state()
+        if isinstance(sensor_state, bool):
+            return sensor_state
+        return sensor_state.lower() in (STATE_ON, "on")
```

Comment:
The idea in `STATE_LOOKUP` is to not consider the value returned by the library but just convert it to a valid HA state. If a new state will come from the library, will be enough add new mapping without changing the function and the logic inside. The final result here of course is the same, but this just a chance. The idea is to keep library and integration totally separated (and in future I will separate)